### PR TITLE
docs: update migration.md to match latest standards

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Migration page had outdated information.

## What is the new behavior?
Updates the page with the latest examples and recommendations via the Beta API.

## Other information
Click the Netlify preview link to see a PR build of the docs.

### CODE REVIEW

File: website/docs/development/getting-started/migration.md [hunk 1]
```diff
@@ -71,99 +77,191 @@ In order to use PD, we need to get it into your project first. For most projects
...
 import { getButtonProps } from '@pluralsight/headless-styles'
 import type { ButtonOptions } from '@pluralsight/headless-styles/types'
 
-// 💡 Keeping logic out of your components improves performance
 const cancelBtnProps = getButtonProps({
   sentiment: 'danger',
 })
 
-export default function CancelButton(props: ButtonOptions) {
+function CancelButton(props: ButtonOptions) {
```
Comment (Rykus0) [inline website/docs/development/getting-started/migration.md:111]: Should this type be their props, and not the ones for Headless-styles?

Comment (caseybaggz) [reply website/docs/development/getting-started/migration.md:111]: @Rykus0 in this example, the point is showing we provide a props type. We don't want to assume all people either have props or are even using TS for that matter. Make sense?

Comment (Rykus0) [reply website/docs/development/getting-started/migration.md:111]: Yeah, I was thinking that the type defines the options for our API (sentiment, etc), and not the props that would be passed to the component (onClick, etc).  I see that the ButtonOptions type also includes the base props for Button, so I think it works in this case, but I don't think we did that everywhere.

**Review by Rykus0 — APPROVED** [review]

File: website/docs/development/getting-started/migration.md (hunk 1 above)
Comment (caseybaggz) [reply website/docs/development/getting-started/migration.md:111]: Nope! Only for the Button API since it matches the use-case more.

**Pull request merged**